src/main/scala/dpla/ingestion3/executors/EnrichExecutor.scala

```diff
@@ -92,33 +101,89 @@ trait EnrichExecutor extends Serializable {
       .format("com.databricks.spark.avro")
       .save(dataOut)
 
+    val endTime = System.currentTimeMillis()
+
+    val typeMessages = messages.filter("field='type'")
+    val dateMessages = messages.filter("field='date'")
+    val langMessages = messages.filter("field='language'")
+    val placeMessages = messages.filter("field='place'")
+
+    val logEnrichedFields = List(
+      "all" -> messages,
+      "type" -> typeMessages,
+      "date" -> dateMessages,
+      "language" -> langMessages,
+      "place" -> placeMessages
+    )
+      .filter { case (_, data: Dataset[_]) => data.count() > 0 }
+
+    val logFileSeq = logEnrichedFields.map {
+      case (name: String, data: Dataset[_]) => {
+        val path = dataOut + "/../logs/"+ s"$shortName-$endTime-enrich-$name"
+        data match {
+          case dr: Dataset[Row] => Utils.writeLogsAsCsv(path, name, dr, shortName)
+        }
+        ReportFormattingUtils.centerPad(name, new File(path).getCanonicalPath)
+      }
+    }
+
+    // build the information for reporting
+    val attemptedCount = mappedRows.count()
+
+    val timeSummary = TimeSummary(
+      Utils.formatDateTime(startTime),
+      Utils.formatDateTime(endTime),
+      Utils.formatRuntime(endTime-startTime)
+    )
+
+    val operationSummary = OperationSummary(
+      attemptedCount,
+      improvedCount.count,
+      attemptedCount-improvedCount.count,
+      logFileSeq
+    )
+
+    val enrichOpSummary = EnrichmentOpsSummary(
+      typeMessages.count,
+      dateMessages.count,
+      langMessages.count,
+      placeMessages.count,
+      langSummary = PrepareEnrichmentReport.generateFieldReport(messages, "language"),
+      typeSummary = PrepareEnrichmentReport.generateFieldReport(messages, "type"),
+      placeSummary = PrepareEnrichmentReport.generateFieldReport(messages, "place")
+    )
+
+    val summaryData = EnrichmentSummaryData(shortName, operationSummary, timeSummary, enrichOpSummary)
+
+    // Write warn and error messages to CSV files
+    logger.info(EnrichmentSummary.getSummary(summaryData))
```

Comment:
Does line 159 really write a CSV file?  It looks like `EnrichmentSummary.getSummary` writes log output, but maybe I'm reading it wrong.